test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/PackageSearchRunnerTests.cs

```diff
@@ -0,0 +1,387 @@
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Test.Utility;
+using Xunit;
+using System.IO;
+using NuGet.Configuration;
+using NuGet.CommandLine.XPlat;
+using System;
+using System.Linq;
+using System.Globalization;
+
+namespace NuGet.CommandLine.Xplat.Tests
+{
+    public class PackageSearchRunnerTests
+    {
+        readonly string _onePackageQueryResult = $@"
+                {{
+                    ""@context"":
+                    {{
+                        ""@vocab"": ""http://schema.nuget.org/schema#"",
+                        ""@base"": ""https://api.nuget.org/v3/registration5-semver1/""
+                    }},
+                    ""totalHits"": 396,
+                    ""data"": [
+                    {{
+                        ""@id"": ""https://api.nuget.org/v3/registration5-semver1/newtonsoft.json/index.json"",
+                        ""@type"": ""Package"",
+                        ""registration"": ""https://api.nuget.org/v3/registration5-semver1/newtonsoft.json/index.json"",
+                        ""id"": ""Fake.Newtonsoft.Json"",
+                        ""version"": ""12.0.3"",
+                        ""summary"": """",
+                        ""title"": ""Json.NET"",
+                        ""iconUrl"": ""https://api.nuget.org/v3-flatcontainer/newtonsoft.json/12.0.3/icon"",
+                        ""licenseUrl"": ""https://www.nuget.org/packages/Newtonsoft.Json/12.0.3/license"",
+
+                        ""tags"": [
+                            ""json""
+                        ],
+
+                        ""authors"": [
+                        ""James Newton-King""
+                        ],
+
+                        ""totalDownloads"": 531607259,
+                        ""verified"": true,
+
+                        ""packageTypes"": [
+                        {{
+                            ""name"": ""Dependency""
+                        }}
+                        ],
+
+                        ""versions"": [
+                        {{
+                            ""version"": ""3.5.8"",
+                            ""downloads"": 461992,
+                            ""@id"": ""https://api.nuget.org/v3/registration5-semver1/newtonsoft.json/3.5.8.json""
+                        }}
+                        ]
+                    }}
+                    ]
+                }}";
+
+        [Fact]
+        public async Task PackageSearchRunner_SearchAPIReturnsOnePackage_OnePackageTableOutputted()
+        {
+            // Arrange
+            ISettings settings = Settings.LoadDefaultSettings(Directory.GetCurrentDirectory(),
+                    configFileName: null,
+                    machineWideSettings: new XPlatMachineWideSetting());
+            PackageSourceProvider sourceProvider = new PackageSourceProvider(settings);
+            var logger = new TestLoggerWithColor();
```

Comment:
I'm trying to steer us toward Mocking rather than creating test classes. Did you happen to try mocking an `ILoggerWithColor`? You could keep the queue abstracted like you've done in a class, but instead create a shared xunit Setup that builds a queue (or list, or whatever data structure). 
Here's an example of a callback where I store the telemetry event. You can imagine how you could accumulate logs as well. (I believe there's an example of that, but I didn't find it quickly)
https://github.com/NuGet/NuGet.Client/blob/6af47d6bf6ceaa0bc99aa1a8e3af0cd759b04a37/test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Actions/UIActionEngineTests.cs#L262

I tend to start with a shared helper method in the class, then if it seems useful later, it's easy to move it to a shared Test Utility class. The goal is that we don't want to copy/paste the same setup when adding new tests later, but it's okay IMO to wait until the need comes up.